coderd/database/pubsub_test.go

```diff
@@ -117,11 +115,17 @@ func TestPubsub_ordering(t *testing.T) {
 	}
 }
 
+// disconnectTestPort is the hardcoded port for TestPubsub_Disconnect.  In this test we need to be able to stop Postgres
+// and restart it on the same port.  If we use an ephemeral port, there is a chance the OS will reallocate before we
+// start back up.  The downside is that if the test crashes and leaves the container up, subsequent test runs will fail
+// until we manually kill the container.
+const disconnectTestPort = 26892
```

Comment:
I hope this, by default, usually falls outside the dynamic port range of `:0`, so we won't collide with other tests/test packages. Are the rules same on Windows?

At least on my NAS, it seems OK:
```
❯ cat /proc/sys/net/ipv4/ip_local_port_range
32768	60999
```

Assuming these values are set differently, there could still be port collisions.

Something like `port := testutil.Port(26892)`, where that function takes the input as _preferred_, but falls back to seeking the closest free port would help in the case mentioned in the comment. The package can store "used ports", unfortunately I don't think that state won't be shared between test packages (could be wrong though).

Reply:
Regarding Windows, we don't run these tests there.  Presumably because Docker on Windows is more effort/overhead than it's worth.

Reply:
Using `testutil.Port()` is clever, but thinking about it a bit, I don't think we should be that clever in tests.  If the test leaks the container, I'd rather us know about it by failing tests, versus doing something different and rare.